commit b4d59784888d34467a8dc782e978901369eedbb0 (refs/changes/18/518/1)
Author: dboldyrev <Arc3111@gmail.com>
Date:   2016-09-20T11:45:30+03:00 (3 years, 1 month ago)
    
    ADMINUI-2334 UI doesn't show errors returned from tags API calls

diff --git a/www/js/components/pages/vm/index.jsx b/www/js/components/pages/vm/index.jsx
index 4da6618d..bc8d0b72 100644
--- a/www/js/components/pages/vm/index.jsx
+++ b/www/js/components/pages/vm/index.jsx
@@ -569,7 +569,20 @@ var VMPage = React.createClass({
                 });
             }
         }
-        vm.updateTags(data, function (job, err) {
+        vm.updateTags(data, function (job, error) {
+            if (error) {
+                var message = error.message || error;
+                if (error.errors) {
+                    var tagsError = error.errors.find(function (errorObject) {
+                        return errorObject.field === 'tags';
+                    });
+                    message = tagsError && tagsError.message || message;
+                }
+                adminui.vent.trigger('notification', {
+                    level: 'error',
+                    message: message
+                });
+            }
             if (job) {
                 self.setState({currentJob: job});
                 job.on('finished', function () {
